pallets/erc20-xcm-bridge/src/erc20_matcher.rs

```diff
@@ -67,6 +67,21 @@ impl<Erc20MultilocationPrefix: Get<MultiLocation>> Erc20Matcher<Erc20Multilocati
 			_ => Err(()),
 		}
 	}
+	pub(crate) fn matches_gas_limit(multiasset: &MultiAsset) -> Option<u64> {
+		let multilocation = match (&multiasset.fun, &multiasset.id) {
+			(Fungible(_), Concrete(ref id)) => id,
+			_ => return None,
+		};
+
+		let prefix = Erc20MultilocationPrefix::get();
+		match multilocation.interior().at(prefix.interior().len() + 1) {
```

Comment:
I based the `prefix.interior().len() + 1` positioning from the `matches_erc20_multilocation` function. Maybe there is a better way of doing it

Reply:
This should be safe because an index-out-of-bounds returns a `None`. However, I had to look at the XCM code and also your test case to discern this.

A comment in the code about these points would go a long way. You could also move `prefix.interior().len() + 1` up and define as:

```rust
let maybe_gas_limit = prefix.interior().len() + 1;
```

Or similar.

Reply:
Please always take the last Junction, you can do that by iterating the reverse order: 

```
let maybe_gas_limit = prefix.interior().into_iter().next_back();
```